### ISSUE TITLE
CPU 101%

### ISSUE THREAD

**Issue by keshavd** [locomojis/Chromoji#11]
Since I installed this extension, a Chrome process that runs over 100% keeps appearing in Activity Monitor.

Disabled it and now I'm back to normal. I believe the extension is the issue.

I'm on Lion 10.7.5, and Chrome Version 25.0.1364.99.

**Comment by RobJDavey** [comment]
A couple of questions:
1. Is this a constant 100%, or just spiking up to 100% every so often and then coming back down again?
2. Which version of Chromoji do you currently have installed?

**Comment by keshavd** [comment]
1. Every so often, it would crank up 100% for about 5 minutes then go back down. In one case I wasn't browsing. I just had Chrome open with no active windows.
2. 1.1.2

**Comment by RobJDavey** [comment]
Thanks for getting back to me with those answers.

That's quite strange. I've not seen this behaviour yet. It's not unreasonable to think there may be issues with Chromoji not playing nicely when combined with a different extension however.
The images are all checked at Chrome startup so I would expect to see some high CPU usage right at the very start and then only see it spike whenever you tried to load a new page. The fact that this is constant surprises me.
At the moment I have no Mac available for testing so I cannot see if this is a Mac specific issue.

Would you be able to **enable all** of the Emoji character sets in the options and visit [this page](http://www.grumdrig.com/emoji-list/) and tell me
1. How long did it take from pressing Enter to the images being displayed?
2. Did all the images display correctly?
3. Is the CPU still high even after the page has fully rendered?

It's also worth noting that if you right click on the top of the Chrome window, you can open the Chrome Task Manager. This might help you to determine if it is the Chromoji extension that's causing the problem, or a different extension. It might be that a different extension is using the high CPU but it is a problem caused by Chromoji. If this is the case we will need to track down which extension is having the issue.

**Comment by keshavd** [comment]
I enabled the plugin and the issue started again but its very sporadic. I cannot recreate the issue on demand.

The issue is characterized by the fan on full power and immediate battery drain. One of the many processes named either Google Chrome Worker or Renderer appears in Activity Monitor and hits over 100%. When I force-close it, the only error I see is one from Chromoji. Once disabled, I do not have the issue.

The extensions I am running alongside it are: Gmelius, After the deadline, adblock, google calender, chrome to phone, google tasks, mightytext and notifier for twitter.  

Plugins running are: Shockwave Flash (and its Broker), Google Talk Plugin

Today's issue I barely caught. I spotted Chromoji at a value of 76 and then 0 in the Google Chrome Task Manager. The fan immediately went silent. This issue lasted a total of about a minute . This was when I pulled down "more stories" from the newsfeed.

Every time I load a new page, Chromoji CPU activity moves from 0 to 104 back to 0 in about 7 seconds.

To answer your questions:
1. It took about 7 seconds.
2. All were displayed correctly
3. Once fully displayed it did drop from 103 to 76 to 0 within a matter of seconds.

Chromoji is the only process I have that has been hitting such high levels, the rest while regularly browsing hit maybe a maximum of 14, and rarely adblock will hit 40.

This has only happened to me once before with a previous application. It kept being a runaway process and once it was uninstalled I never heard my fan again.

EDIT: A few are Emoji are missing:
Squared Katana KOKO to Squared CJK Unified Ideograph-71221. SQUARED CJK UNIFIED IDEOGRAPH-6708, SQUARED CJK UNIFIED IDEOGRAPH-7533, SQUARED KATAKANA SA, CIRCLED IDEOGRAPH ACCEPT, TRADE MARK SIGN

EDIT: I did catch it! Here's the chrome task manager snip from when the issue acted up
![Screen Shot 2013-03-01 at 1 04 57 AM](https://f.cloud.github.com/assets/3721736/208618/61b01874-823c-11e2-88fb-dc1c8239abe4.png)

**Comment by RobJDavey** [comment]
It sounds like this issue may just be entirely caused by the _extremely_ intensive work it has to do to insert the images correctly. I'm constantly working to improve this performance however and I have a few ideas for a future release _(that will rely on #8 being completed first)_ that will hopefully speed up the process a lot.

I'm hoping to do a bit more either tonight or over the weekend so I'll let you know if I make any progress on the performance front.

I test this on an _8 year old_ machine at home so I was hoping the performance would be OK but seems like there's still some work to do yet!

I'm not sure how to explain your missing symbols though. That is strange. It may be a problem with the messaging channels I use to communicate with the settings object running on the background page but that seems unlikely. Again though, #8 is a complete re-write of a large part of the system so I can support the [SoftBank Encoding](http://en.wikipedia.org/wiki/Emoji#SoftBank_Encoding) (iOS 5). This may change the behaviour a lot so we can check again once this is done.

I'm sorry I don't have an instant solution at this point in time but I'll do my best to resolve this (if it still happens) when [1.2](https://github.com/robjdavey/Chromoji/issues?milestone=3) is released.

**Comment by RobJDavey** [comment]
Good news @Ultimawar! The changes I'm making for 1.2 seem **a lot** faster! This will hopefully fix your CPU issues. They're so fast I'm getting rid of the settings page because it's fine with every single Emoji enabled!

**Comment by keshavd** [comment]
That's awesome! I can't wait to try it out :)

**Comment by RobJDavey** [comment]
1.2 is now released. Hopefully this will fix your issues. If not please raise it as a new issue. Thanks for your help and input!

**Comment by keshavd** [comment]
No issues with the new version! :)

**Comment by RobJDavey** [comment]
That's great to hear @Ultimawar! Is the performance OK for you too? It should be much faster at displaying the Emoji.

**Comment by keshavd** [comment]
With the new version the performance is perfect, I can't tell I'm running an extension. The emoji literally load instantly.
